src/mlpack/tests/lmnn_test.cpp

```diff
@@ -235,51 +232,51 @@ BOOST_AUTO_TEST_CASE(LMNNSeparableGradientTest)
 
   lmnnfn.Gradient(coordinates, 0, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
 
   lmnnfn.Gradient(coordinates, 1, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
 
   lmnnfn.Gradient(coordinates, 2, gradient, 1);
 
-  BOOST_REQUIRE_CLOSE(gradient(0, 0), -0.048, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(0, 1), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 0), 0.0, 1e-5);
-  BOOST_REQUIRE_CLOSE(gradient(1, 1), 2.0, 1e-5);
+  REQUIRE(gradient(0, 0) == Approx( -0.048).epsilon(1e-7));
+  REQUIRE(gradient(0, 1) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 0) == Approx( 0.0).epsilon(1e-7));
+  REQUIRE(gradient(1, 1) == Approx( 2.0).epsilon(1e-7));
```

Comment:
```suggestion
  REQUIRE(gradient(0, 0) == Approx(-0.048).epsilon(1e-7));
  REQUIRE(gradient(0, 1) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 0) == Approx(0.0).epsilon(1e-7));
  REQUIRE(gradient(1, 1) == Approx(2.0).epsilon(1e-7));
```

Remove the extra space.